pwiz_tools/Skyline/Model/MSAmanda/MSAmandaSearchWrapper.cs

```diff
@@ -0,0 +1,315 @@
+
+            ////// set searchagainstdecoy
+            ////_searchAgainstDecoy = _settings.GenerateDecoyDb;
+            //// save fasta with numbers in own folder                        
+            //GenerateMappings();
+            //// Generate Directories for SearchEngine
+            //GenerateDirectoriesForSearchEngine();
+        }
+
+        //private void GenerateMappings()
+        //{
+        //    var folderForMappings = Path.Combine(_baseDir, AmandaMap);
+        //    //if (!_settings.FolderName.ToUpper().Equals("DEFAULT"))
+        //    //{
+        //    //    folderForMappings = Path.Combine(_settings.FolderName, MsAmanda2Foldername, AmandaMap);
+        //    //}
+
+        //    // creates dir if not existing
+        //    Directory.CreateDirectory(folderForMappings);
+        //    // for normal search
+        //    // get correct filenames            
+        //    _mappedFastaFile = _helper.GetFastaMapFileName(folderForMappings, _outputParameters.DBFile);
+        //    // mappingfile, internal ids to geneidentifiers
+        //    _mappingFile = _helper.GetFastaMappingFileName(folderForMappings, _outputParameters.DBFile);
+        //    // check if files exist and are generated before last changedate from dbFileName
+        //    _helper.CheckForMappingFile(_outputParameters.DBFile, _mappedFastaFile, _mappingFile, _outputMzid);
+
+        //    // for decoy search
+        //    // IDs with REV_
+        //    // filenames with _map_decoy   
+        //    if (_settings.PerformDecoySearch)
+        //    {
+        //        _decoyMappedFastaFile = _helper.GetFastaMapFileName(folderForMappings, _outputParameters.DBFile, true);
+        //        // mappingfile, internal ids to geneidentifiers
+        //        _decoyMappingFile = _helper.GetFastaMappingFileName(folderForMappings, _outputParameters.DBFile, true);
+        //        // check if files exist and are generated before last changedate from dbFileName
+        //        _helper.CheckForMappingFile(_outputParameters.DBFile, _decoyMappedFastaFile, _decoyMappingFile,
+        //            _outputMzid, true);
+        //    }
+        //}
+
+        private bool success = true;
+        public override bool Run(CancellationTokenSource tokenSource)
+        {
+            try
+            {
+                InitializeEngine(tokenSource);
+                List<Spectrum> spectra = new List<Spectrum>();
+                foreach (string rawFileName in SpectrumFileNames)
+                {
+                    if (tokenSource.Token.IsCancellationRequested)
+                    {
+                        
+                        tokenSource.Token.ThrowIfCancellationRequested();
+                    }
+
+                    amandaInputParser = new MSAmandaSpectrumParser(rawFileName, Settings.ConsideredCharges, true);
+                    SearchEngine.SetInputParser(amandaInputParser);
+
+                    SearchEngine.PerformSearch(_outputParameters.DBFile);
```

Comment:
I don't really understand what's going on when I give it multiple files to search. After calling this function on the first file in SpectrumFileNames, MSAmanda apparently calls GetTotalNumberOfSpectra() on all the files in SpectrumFileNames. This seems to be a bug in MSAmanda.